src/btree/bt_sync.c

```diff
@@ -109,44 +109,53 @@ __sync_ref_is_obsolete(WT_SESSION_IMPL *session, WT_REF *ref)
 {
     WT_ADDR *addr;
     WT_CELL_UNPACK vpack;
+    WT_PAGE_MODIFY *mod;
 
     /* Ignore root pages as they can never be deleted. */
     if (__wt_ref_is_root(ref))
         return (false);
 
-    /* Guard against marking an in-memory page to be deleted. */
-    if (ref->state != WT_REF_DISK)
-        return (false);
-
     /* Ignore internal pages, these are taken care of during reconciliation. */
     if (!__wt_ref_is_leaf(session, ref))
         return (false);
 
+    /* In-memory ref address must be from an off page address. */
+    mod = (ref->page != NULL) ? (ref->page->modify) : NULL;
+    if (mod != NULL && mod->rec_result == WT_PM_REC_REPLACE)
+        return (__wt_txn_visible_all(
+          session, mod->mod_replace.newest_stop_txn, mod->mod_replace.newest_stop_ts));
+
+    if (mod != NULL && mod->rec_result == WT_PM_REC_MULTIBLOCK)
+        return (__wt_txn_visible_all(
+          session, mod->mod_multi_newest_stop_txn, mod->mod_multi_newest_stop_ts));
+
```

Comment:
I don't strongly object to this, but I don't see much point in it either. Normal eviction mechanisms should find and evict this page, so I think it's an unnecessary complication to try and push in-memory pages out during the checkpoint. Maybe it's necessary, but we don't yet have any data (of which I'm aware), that shows that to be true.

Reply:
I feel checkpoint adding the obsolete pages to the eviction queue will improve the cache usage than eviction server picking the pages. Eviction server works on picking random pages, so it is not guaranteed that the obsolete pages are cleaned as it is done by the checkpoint.

Lets wait to here others opinions and then we can decide about it.

Reply:
I would prefer simplicity and let eviction do eviction and keep this as something we can do if necessary. Simpler code is better to start.
If we keep this, the comment needs to be clearer. The code is subtle in the fields it is using that are only different in the middle of a long name and dereference.

Reply:
As discussed, Currently I am going ahead with checkpoint adding the pages to urgent evict queue. Later we can revisit this decision if any problem arises.